keps/sig-node/5526-pod-level-resource-managers/README.md

```diff
@@ -0,0 +1,1741 @@
+supported for pod-level resources.
+
+-   **`none`**: This policy makes the Topology Manager idle. No changes are
+    needed.
+-   **`static`**: This policy enables the CPU manager functionality, and
+    contains all logic for the policy behavior, including alignment and
+    allocation, which need to be updated to support pod level resources.
+
+#### Policy Options
+
+-   **`distribute-cpus-across-numa`**: Distributing CPUs across NUMA nodes will
+    have the same behavior as the one for container level resources, the total
+    number of pod level CPUs will be evenly distributed (using modulo, any
+    remaining will be stripped across the NUMA nodes subset) across all
+    available NUMA nodes. It is important to mention that this policy focuses on
+    the distribution, not in CPU exclusivity, facilitating parallel algorithms
+    to run more efficiently.
+-   **`distribute-cpus-across-cores`**: This policy is analogous to the
+    distribute-cpus-across-numa policy option, because of spreading the CPU
+    allocations out, rather than packing them together. However, in this case
+    the total number of pod level CPUs will be spread across cores, and this
+    policy relies on changing the sorting algorithm for sockets, cores and CPUs.
+    This policy has the purpose of leveraging L2 Cache.
+-   **`full-pcpus-only`**: This policy is highly beneficial for multi-tenant
+    pods requiring inter-pod isolation, as it helps prevent hyperthread
+    contention. The pod level CPUs will be aligned and allocated in full
+    physical cores, with a shared CPU pool for its containers.
+-   **`align-by-socket`**: This policy ensures all a pod's CPUs remain on the
+    same socket when possible, reducing inter-socket latencies and benefiting
+    containers that share L3 cache or communicate frequently. The pod level CPU
+    pool will be aligned and allocated in the same socket.
+-   **`strict-cpu-reservation`**: This policy is compatible and crucial for
+    guaranteed workloads, preventing interference from burstable and best-effort
+    pods. The pod level CPUs will be reserved for the overall pod, creating a
+    shared, reserved and isolated CPU pool for the containers.
+-   **`prefer-align-cpus-by-uncorecache`**: This policy optimizes CPU allocation
+    across uncore cache groups, enhancing shared cache locality for the overall
+    pod level CPU pool.
+
+#### Pod Scope Allocation and Partitioning Algorithm
+
+The following algorithm describes the process for the CPU Manager. The Memory
+Manager follows a similar process.
+
+1.  **One-Time Pod-Level Calculation:** When the `Allocate` function is called
+    for the first container of a pod, the manager performs a one-time
+    calculation for the entire pod. It determines the total pod resource budget
+    from `pod.spec.resources` and uses the Topology Manager's hint to identify
+    the single, NUMA-aligned set of CPUs that will be managed for the pod.
```

Comment:
NUMA alignment is not always guaranteed.

Reply:
I added a explicit mention for that.